logging/interceptor.go

```diff
@@ -39,6 +53,109 @@ func LogLevelInterceptor(defaultLevel logrus.Level) grpc.UnaryServerInterceptor
 	}
 }
 
+func ClientInterceptor(logger *logrus.Logger, opts ...Option) grpc.UnaryClientInterceptor {
+	o := evaluateClientOpt(opts)
+	return func(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
+		startTime := time.Now()
+		fields := newLoggerFields(method, startTime, clientKind)
+		err := invoker(ctx, method, req, reply, cc, opts...)
+		addSpecificFields(ctx, fields, logger, o)
+		code := o.codeFunc(err)
+		level := o.levelFunc(code)
+		durField, durVal := o.durationFunc(time.Since(startTime))
+		fields["grpc.code"] = code.String()
+		fields[durField] = durVal
+
+		if err != nil {
+			fields[logrus.ErrorKey] = err
+		}
+
+		levelLogf(
+			logrus.NewEntry(logger).WithFields(fields),
+			level,
+			"finished unary call with code "+code.String())
+		return err
+	}
+}
+
+func ServerInterceptor(logger *logrus.Logger, opts ...Option) grpc.UnaryServerInterceptor {
+	o := evaluateServerOpt(opts)
+	return func(ctx context.Context, req interface{}, info *grpc.UnaryServerInfo, handler grpc.UnaryHandler) (interface{}, error) {
+		startTime := time.Now()
+		fields := newLoggerFields(info.FullMethod, startTime, serverKind)
+		newCtx := newLoggerForCall(ctx, logrus.NewEntry(logger), fields)
+
+		resp, err := handler(newCtx, req)
+
+		if !o.shouldLog(info.FullMethod, err) {
+			return resp, err
+		}
+
+		addSpecificFields(ctx, fields, logger, o)
+		code := o.codeFunc(err)
+		level := o.levelFunc(code)
+		durField, durVal := o.durationFunc(time.Since(startTime))
+		fields["grpc.code"] = code.String()
+		fields[durField] = durVal
+
+		if err != nil {
+			fields[logrus.ErrorKey] = err
+		}
+
+		levelLogf(
+			ctxlogrus.Extract(newCtx).WithFields(fields),
+			level,
+			"finished unary call with code "+code.String())
+
+		return resp, err
+	}
+}
+
+func addSpecificFields(ctx context.Context, fields logrus.Fields, logger *logrus.Logger, o *options) {
+	reqID, exists := requestid.FromContext(ctx)
+	if !exists || reqID == "" {
+		reqID = uuid.New().String()
+	}
+	fields[requestid.DefaultRequestIDKey] = reqID
+	ctx = metadata.AppendToOutgoingContext(ctx, requestid.DefaultRequestIDKey, reqID)
+
+	if accountID, err := auth.GetAccountID(ctx, nil); err == nil {
+		fields[auth.MultiTenancyField] = accountID
+		ctx = metadata.AppendToOutgoingContext(ctx, auth.MultiTenancyField, accountID)
+	} else {
+		logger.Errorf("Unable to get %s from context", auth.MultiTenancyField)
+	}
+
+	if o.ophIDEnabled {
```

Comment:
this entire block is implementation specific and does not belong in the opensource project. I think you want to implement ways to instruct the toolkit to parse out specific fields, namely ophid. I believe this field will become less useful in the future as account ID is included ( in all cases ) for a JWT

Reply:
Had the same thoughts regarding the ophid, but according to docs I understood this like we need logger with specific fields, set by default and not customizable.